0.1/607 setup laravel data models

charlesstrange2 opened the pull request:
Models, migrations and relationships.

charlesstrange2 committed:
Generate initial models
Create initial migrations.

charlesstrange2 committed:
Add intial relationships

charlesstrange2 committed:
Centre class name typo

charlesstrange2 committed:
Establish foreign keys
Fix table creation order (sponsors->centres->users)

katjam commented:
As a general rule - I don't add any relationship methods until I need to use them That saves on unused code and unnecessary test writing.

katjam commented on database/migrations/2017_12_04_113055_create_children_table.php:
```diff
@@ -0,0 +1,38 @@
+        Schema::create('children', function (Blueprint $table) {
+            $table->increments('id');
+            $table->dateTime('dob');
+            $table->boolean('born')->default(false);
```
Could this be appended? If there is a bith date in the past then yes! If not, no! ?

charlesstrange2 replied:
It's a flag to say "this person has been acknowledged as actually being 'born'". 
While the future DoB was considered. They wanted a flag to say "we've been told this person has been born". There may be a UI/Print notice to check with the family that.

katjam commented on database/migrations/2017_12_04_113032_create_families_table.php:
```diff
@@ -0,0 +1,33 @@
+    {
+        Schema::create('families', function (Blueprint $table) {
+            $table->increments('id');
+            $table->string('rvid')->nullable();
```
Maybe a comment // Rose Vouchers Id

charlesstrange2 replied:
Agreed. is not obvious.

katjam commented on database/migrations/2017_12_04_113404_create_notes_table.php:
```diff
@@ -0,0 +1,42 @@
+            $table->increments('id');
+            $table->text('content'); // 64k
+            $table->integer('family_id')->unsigned(); // FK Families
+            $table->integer('user_id')->unsigned(); // FK Users
```
Is this the creator of the note?

charlesstrange2 replied:
Yes. Associates a user with the note.

katjam reviewed (approved):
Probably OK - but I'd think hard about not having the relationship methods until you need them... maybe comment out for now and when you finish this round delete the ones you did not need.

charlesstrange2 committed:
Merge remote-tracking branch 'origin/0.1/release' into 0.1/607-setup-laravel-data-models

charlesstrange2 committed:
Commented for clarity
Changes assumption on "born" folks to be true;
Code should force born false if DoB in future.
